.github/workflows/master-push-pull.yaml

```diff
@@ -0,0 +1,76 @@
+name: Run tests
+
+on:
+  push:
+  pull_request:
+    branches:
+      - main
+
+
+
+jobs:
+  Unit-Tests:
+    runs-on: ubuntu-latest
+    name: Tests
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        name: Setup cache
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Set up node
+        uses: actions/setup-node@v2.4.1
+        with:
+          node-version: '16'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Set ENV
+        run: mv .env.example .env
+
+      - name: Build application
+        run: npm run build
+
+      - name: Run tests
+        run: npm test
+
+
+  TypeScript-Static-Analyis:
+    runs-on: ubuntu-latest
+    name: TypeScript static analysis
+    needs: Unit-Tests
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        name: Setup cache
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Set up node
+        uses: actions/setup-node@v2.4.1
+        with:
+          node-version: '16'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Set ENV
+        run: mv .env.example .env
```

Comment:
suggestion: create a `.env` file specifically for the CI.

Repurposing the example for use by the CI, is unlikely to be implicitly known by people accessing the repo, so changing the example and having the CI break ....is unintuitive.

They may be a convention about having a `.env` file per an environment, but I'm not sure 🤷